cluster/gce/upgrade.sh

```diff
@@ -328,35 +328,50 @@ node_prereqs=false
 local_binaries=false
 env_os_distro=false
 
-while getopts ":MNPlho" opt; do
-  case ${opt} in
-    M)
+if [[ -n $(echo $OSTYPE | grep "darwin") ]]; then
+  # OS X version
+  args=$(getopt MNPlho $@)
```

Comment:
Yeah, this is the portable version. I spent a while looking when I first wrote the script.

I don't like this OS-level check, it makes the script behave differently on OS X than Linux for little purpose?